python/src/aiconfig/editor/server/server.py

```diff
@@ -49,137 +88,120 @@ def home():
-        )
-        return res_save.unwrap_or(
-            HttpResponseWithAIConfig(
-                #
-                message="Failed to save to disk",
-                code=400,
-                aiconfig=None,
-            )
-        ).to_flask_format()
+    aiconfig = state.aiconfig
+    request_json = request.get_json()
+
+    def _op(aiconfig: AIConfigRuntime, op_args: OpArgs) -> Result[None, str]:
+        method_name = MethodName("save")
+        return safe_run_aiconfig_method(aiconfig, method_name, op_args)
+
+    path_val = _validated_request_path(request_json, allow_create=True)
+    op_args: Result[OpArgs, str] = result.do(
+        Ok(OpArgs({"path": path_val_ok}))
+        #
+        for path_val_ok in path_val
+    )
+
+    return run_aiconfig_operation_with_op_args(aiconfig, "save", _op, op_args)
 
 
 @app.route("/api/create", methods=["POST"])
 def create() -> FlaskResponse:
     state = get_server_state(app)
-    state.aiconfig = AIConfigRuntime.create()  # type: ignore
-    return HttpResponseWithAIConfig(message="Created new AIConfig", aiconfig=state.aiconfig).to_flask_format()
+    aiconfig = safe_run_aiconfig_static_method(MethodName("create"), OpArgs({}), AIConfigRuntime)
+    match aiconfig:
+        case Ok(aiconfig_ok):
+            state.aiconfig = aiconfig_ok
+            return HttpResponseWithAIConfig(message="Created new AIConfig", aiconfig=aiconfig_ok).to_flask_format()
+        case Err(e):
+            return HttpResponseWithAIConfig(message=f"Failed to create AIConfig: {e}", code=400, aiconfig=None).to_flask_format()
 
 
 @app.route("/api/run", methods=["POST"])
 async def run() -> FlaskResponse:
     state = get_server_state(app)
+    aiconfig = state.aiconfig
     request_json = request.get_json()
+
+    def _op(aiconfig: AIConfigRuntime, op_args: OpArgs) -> Result[None, str]:
+        method_name = MethodName("run")
+        return safe_run_aiconfig_method(aiconfig, method_name, op_args)
+
     prompt_name = request_json.get("prompt_name", None)
     stream = request_json.get("stream", True)
     LOGGER.info(f"Running prompt: {prompt_name}, {stream=}")
     inference_options = InferenceOptions(stream=stream)
-    try:
-        result = await state.aiconfig.run(prompt_name, options=inference_options)  # type: ignore
-        LOGGER.debug(f"Result: {result=}")
-        return HttpResponseWithAIConfig(
-            message="Done",
-            aiconfig=state.aiconfig,
-        ).to_flask_format()
-    except Exception as e:
-        err: Err[str] = core_utils.ErrWithTraceback(e)
-        LOGGER.error(f"Failed to run: {err}")
-        return HttpResponseWithAIConfig(
-            message=f"Failed to run: {err}",
-            code=400,
-            aiconfig=None,
-        ).to_flask_format()
+
+    op_args = Ok(OpArgs({"prompt_name": prompt_name, "options": inference_options}))
+
+    return run_aiconfig_operation_with_op_args(aiconfig, "run", _op, op_args)
```

Comment:
should this be MethodName("run")?

Reply:
So I distinguish between "method" and more general "operation". The former is for cases where we just forward the arguments to the corresponding method, and the latter is used in cases where the API endpoint has to do more logic. In this case, the extra logic is building the InferenceOptions. I can see why this is confusing. Suggestions for simplifying?

Reply:
It could technically be simplified down to one method and just always parse out the JSON arguments into op args (if there's a clean way to do that, e.g. `OpArgs({...request_json})`

Reply:
>  just always parse out the JSON arguments into op args 

Sometimes this requires logic specific to the endpoint, like checking that a string is a readable fs path. We can pass unvalidated stuff down into AIConfig and let it raise, but then root causes and response error messages will be less clear.